### DISCUSSION TITLE
Load thousands of files in R

### DISCUSSION THREAD

**Discussion by christophe-rd** [rstudio/ggcheck#38] [Q&A]
HI!

I need to upload about 21 000 csv files from my computer's directory.
They are structured like so: 2005_4.csv where 2005 is the year and 4 is the day of the year. I have those files for more than 20 years. Obviously, I can't load them all one by one and have no idea how to do that operation.
They are all structured the same way though like so:

>structure(list(A08B08_...79.4945...48.4148..296.....2.68 = c("A08E31_ -79.4679 48.4254 278. 2.77",
"A09A03_ -79.4848 48.4178 301. 2.66", "A12B01_ -79.4944 48.4146 296. 2.69",
"A12B01_ -79.4944 48.4146 296. 2.69", "A12B01_ -79.4944 48.4146 296. 2.69",
"B12RJ52 -78.6642 49.4051 275. 1.72", "B12YE02 -78.8871 49.2079 335. 1.87",
"ESB14M3 -79.1944 48.4529 277. 2.83", "F04F51_ -79.2888 48.4951 299. 2.71",
"F09B02_ -79.3472 48.4472 273. 2.81", "F09J11_ -79.3461 48.4964 273. 2.80",
"F12I01_ -79.1944 48.4496 280. 2.82", "F18P01_ -79.2628 48.4395 273. 2.83",
"H05Z02_ -79.3870 48.5190 273. 2.78", "H08Z03_ -79.3712 48.5175 273. 2.79",
"H08Z09_ -79.2122 48.4836 272. 2.85", "H13Z07_ -79.4915 48.5238 302. 2.62",
"H13Z08_ -79.4914 48.5239 302. 2.62", "H13Z25_ -79.2107 48.4825 272. 2.85",
"H18Z27_ -79.4014 48.5175 284. 2.73", "SC14F10 -79.4684 48.4269 278. 2.77",
"A12B01_ -79.4944 48.4146 296. 2.69", "F04F51_ -79.2888 48.4951 299. 2.71",
"F09Y06_ -79.3442 48.4690 276. 2.79", "H05Z02_ -79.3870 48.5190 273. 2.78",
"J20A14_ -79.3595 48.5011 275. 2.79", "J20F05_ -79.3220 48.5038 272. 2.81",
"J21A01_ -79.4791 48.4201 290. 2.71", "J21A06_ -79.3511 48.5180 273. 2.79",
"J21F16_ -79.3619 48.4413 274. 2.80", "F03K08_ -79.2996 48.4407 305. 2.68",
"J22A02_ -79.4816 48.4229 290. 2.71", "J22A12_ -79.3709 48.5176 273. 2.79",
"J22A13_ -79.3526 48.4922 274. 2.80", "J22A36_ -79.3838 48.4229 298. 2.70",
"J22A43_ -79.4866 48.4452 291. 2.70", "J22A50_ -79.4303 48.4779 356. 2.42",
"J22A66_ -79.3386 48.3582 303. 2.68", "J22F05_ -79.3455 48.4663 272. 2.81",
"J22F10_ -79.2696 48.4380 272. 2.83", "J22F11_ -79.3005 48.4410 300. 2.70",
"J20F05_ -79.3222 48.5037 272. 2.81", "S22SD01 -73.8309 45.5396 37. 3.00",
"S22BP01 -73.6831 45.6040 39. 3.07", "S22PP01 -73.5205 45.6910 16. 2.97",
"S22SE01 -73.6260 46.2881 303. 0.54", "S22SE02 -73.6192 46.3146 346. 0.43",
"S22SC01 -73.7259 46.2531 259. 0.55", "S22NO01 -75.0557 46.4020 287. -0.12",
"O22HM33 -79.9951 46.1177 212. 1.35")), class = "data.frame", row.names = c(NA,
-50L))

What I am aiming to do: Create an empty data frame with the four columns mentioned above, and add columns corresponding to the fifth column of each file, where each column would be named according to the corresponding file. I'de like to create a new data frame for each file. 
Hope I was clear enough. Hope someone can help me as I am a little lost!

**Comment by gadenbuie** [comment]
Hi @christophe-rd, I think you'll very likely get better help if you post this question on Posit Community: https://community.rstudio.com/
